MdeModulePkg/Library/UefiBootManagerLib/BmConsole.c

```diff
@@ -206,9 +206,9 @@ EfiBootManagerGetGopDevicePath (
           DEBUG ((DEBUG_INFO, "[Bds] Looking for GOP child deeper ... \n"));
           TempDevicePath   = GopPool;
           ReturnDevicePath = EfiBootManagerGetGopDevicePath (OpenInfoBuffer[Index].ControllerHandle);
-          GopPool          = AppendDevicePathInstance (GopPool, ReturnDevicePath);
-
+          // MU_CHANGE verify ReturnDevicePath is valid before Appending
           if (ReturnDevicePath != NULL) {
+            GopPool = AppendDevicePathInstance (GopPool, ReturnDevicePath);

```

Comment:
I thought `AppendDevicePathInstance` takes NULL input for the second argument?

Reply:
AppendDevicePathInstance does accept a NULL as the second argument, but when NULL is passed for the second argument, NULL is returned:
https://github.com/microsoft/mu_basecore/blob/release/202208/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLibOptionalDevicePathProtocol.c#L203

Given that this is going through all the child protocols of this parent handle, there is the possibility that there are multiple gop child devices. It seems that making the entire pool of device paths NULL could lose any previous appended handles.